src/status_im/ui/screens/ens/views.cljs

```diff
@@ -0,0 +1,511 @@
+                    | (on-chain, 1 tx)                         | (on-chain, 1tx per info to connect)
+                    |                                          |
+                    +-----------------------+------------------+
+                                            |
+                                            |
+                                            |  Saving
+                                            |
+                                            |
+                                    +-------+-----+
+                                    |    Saved    |
+                                    +-------------+
+
+  "
+  (:require [re-frame.core :as re-frame]
+            [reagent.core :as reagent]
+            [status-im.ens.core :as ens]
+            [status-im.ethereum.core :as ethereum]
+            [status-im.ethereum.stateofus :as stateofus]
+            [status-im.i18n :as i18n]
+            [status-im.react-native.resources :as resources]
+            [status-im.ui.components.checkbox.view :as checkbox]
+            [status-im.ui.components.colors :as colors]
+            [status-im.ui.components.common.common :as components.common]
+            [status-im.ui.components.icons.vector-icons :as vector-icons]
+            [status-im.ui.components.list.views :as list]
+            [status-im.ui.components.react :as react]
+            [status-im.ui.components.status-bar.view :as status-bar]
+            [status-im.ui.components.toolbar.actions :as actions]
+            [status-im.ui.components.toolbar.view :as toolbar])
+
+  (:require-macros [status-im.utils.views :as views]))
+
+;; Components
+
+(defn- button [{:keys [on-press] :as m} label]
+  [components.common/button (merge {:button-style {:margin-vertical    8
+                                                   :padding-horizontal 32
+                                                   :justify-content    :center
+                                                   :align-items        :center}
+                                    :on-press     on-press
+                                    :label        label}
+                                   m)])
+
+(defn- link [{:keys [on-press]} label]
+  [react/touchable-opacity {:on-press on-press}
+   [react/text {:style {:text-align-vertical :center :color colors/blue}}
+    label]])
+
+(defn- section [{:keys [title content]}]
+  [react/view {:style {:margin-horizontal 16 :align-items :flex-start}}
+   [react/text {:style {:color colors/gray :font-size 15}}
+    title]
+   [react/view {:margin-top 8 :padding-horizontal 16 :padding-vertical 12 :border-width 1 :border-radius 12
+                :border-color colors/gray-lighter}
+    [react/text {:style {:font-size 15}}
+     content]]])
+
+;; Name details
+
+(views/defview name-details []
+  (views/letsubs [{:keys [name]} [:get-screen-params :ens-name-details]]
+    (let [{:keys [address public-key]} @(re-frame/subscribe [:ens.name/screen name])]
```

Comment:
better to have an extra subscription for this , which will be subscribed on [:get-screen-params :ens-name-details]

Reply:
:+1: